store/src/main/java/org/apache/rocketmq/store/rocksdb/MessageRocksDBStorage.java

```diff
@@ -0,0 +1,652 @@
+    private static final byte[] END_SUFFIX_BYTES = new byte[512];
+    static {
+        Arrays.fill(END_SUFFIX_BYTES, (byte) 0xFF);
+    }
+    private static final Set<byte[]> COMMON_CHECK_POINT_KEY_SET_FOR_TIMER = new HashSet<>();
+    public static final byte[] SYS_TOPIC_SCAN_OFFSET_CHECK_POINT = "sys_topic_scan_offset_checkpoint".getBytes(StandardCharsets.UTF_8);
+    public static final byte[] TIMELINE_CHECK_POINT = "timeline_checkpoint".getBytes(StandardCharsets.UTF_8);
+    static {
+        COMMON_CHECK_POINT_KEY_SET_FOR_TIMER.add(SYS_TOPIC_SCAN_OFFSET_CHECK_POINT);
+        COMMON_CHECK_POINT_KEY_SET_FOR_TIMER.add(TIMELINE_CHECK_POINT);
+    }
+    private static final byte[] DELETE_VAL_FLAG = new byte[] {(byte)0xFF};
+    private static final int LAST_OFFSET_PY_LENGTH = LAST_OFFSET_PY.length;
+
+    private volatile ColumnFamilyHandle timerCFHandle;
+    private volatile ColumnFamilyHandle transCFHandle;
+
+    private final ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+    private static final Cache<byte[], byte[]> DELETE_KEY_CACHE_FOR_TIMER = CacheBuilder.newBuilder()
+        .maximumSize(10000)
+        .expireAfterWrite(60, TimeUnit.MINUTES)
+        .build();
+
+    public MessageRocksDBStorage(MessageStoreConfig messageStoreConfig) {
+        super(Paths.get(messageStoreConfig.getStorePathRootDir(), ROCKSDB_MESSAGE_DIRECTORY).toString());
+        this.start();
+    }
+
+    @Override
+    protected boolean postLoad() {
+        try {
+            UtilAll.ensureDirOK(this.dbPath);
+            initOptions();
+            ColumnFamilyOptions indexCFOptions = RocksDBOptionsFactory.createIndexCFOptions();
+            ColumnFamilyOptions timerCFOptions = RocksDBOptionsFactory.createTimerCFOptions();
+            ColumnFamilyOptions transCFOptions = RocksDBOptionsFactory.createTransCFOptions();
+            this.cfOptions.add(indexCFOptions);
+            this.cfOptions.add(timerCFOptions);
+            this.cfOptions.add(transCFOptions);
+
+            List<ColumnFamilyDescriptor> cfDescriptors = new ArrayList<>();
+            cfDescriptors.add(new ColumnFamilyDescriptor(RocksDB.DEFAULT_COLUMN_FAMILY, indexCFOptions));
+            cfDescriptors.add(new ColumnFamilyDescriptor(TIMER_COLUMN_FAMILY, timerCFOptions));
+            cfDescriptors.add(new ColumnFamilyDescriptor(TRANS_COLUMN_FAMILY, transCFOptions));
+            this.open(cfDescriptors);
+            this.defaultCFHandle = cfHandles.get(0);
+            this.timerCFHandle = cfHandles.get(1);
+            this.transCFHandle = cfHandles.get(2);
+            scheduler.scheduleAtFixedRate(() -> {
+                try {
+                    db.flush(flushOptions, timerCFHandle);
+                    log.info("MessageRocksDBStorage flush timer wal success");
+                } catch (Exception e) {
+                    logError.error("MessageRocksDBStorage flush timer wal failed, error: {}", e.getMessage());
```

Comment:
Directly printing the exception is preferable, as printing e.getMessage() may lose important stack trace information, hindering troubleshooting.